atef/widgets/config/run_base.py

```diff
@@ -0,0 +1,416 @@
+    prepared_file: PreparedFile,
+    original_c: Union[Configuration, Comparison]
+) -> List[Union[PreparedConfiguration, PreparedComparison]]:
+    """
+    Gather all the PreparedConfiguration or PreparedComparison dataclasses
+    that correspond to the original comparison or config.
+
+    Phrased another way: maps prepared comparisons onto the comparison
+    seen in the GUI
+
+    Currently for passive checkout files only
+
+    Parameters
+    ----------
+    prepared_file : PreparedFile
+        the file containing configs or comparisons to be gathered
+    original_c : Union[Configuration, Comparison]
+        the comparison to match PreparedComparison or PreparedConfigurations to
+
+    Returns
+    -------
+    List[Union[PreparedConfiguration, PreparedComparison]]:
+        the configuration or comparison dataclasses related to ``original_c``
+    """
+    matched_c = []
+
+    for config in prepared_file.walk_groups():
+        if config.config == original_c:
+            matched_c.append(config)
+
+    for comp in prepared_file.walk_comparisons():
+        if comp.comparison == original_c:
+            matched_c.append(comp)
+
+    return matched_c
+
+
+class RunCheck(DesignerDisplay, QWidget):
+    """
+    Widget to be added to run widgets
+
+    Connections: (to establish)
+    - next button
+    - verify button to pop-out and record
+
+    Parent widget must be a PageWidget
+    """
+    filename = 'run_check.ui'
+
+    # Left to right
+    result_label: QLabel
+    result_verify_spacer: QSpacerItem
+    verify_button: QToolButton
+    verify_label: QLabel
+    verify_run_spacer: QSpacerItem
+    run_button: QPushButton
+    run_success_label: QLabel
+    run_next_spacer: QSpacerItem
+    next_button: QPushButton
+
+    style_icons = {
+        Severity.success: QStyle.SP_DialogApplyButton,
+        Severity.warning : QStyle.SP_TitleBarContextHelpButton,
+        Severity.internal_error: QStyle.SP_DialogCancelButton,
+        Severity.error: QStyle.SP_DialogCancelButton
+    }
+
+    unicode_icons = {
+        # check mark
+        Severity.success: '<span style="color: green;">&#10004;</span>',
+        Severity.warning : '<span style="color: orange;">?</span>',
+        # x mark
+        Severity.internal_error: '<span style="color: red;">&#10008;</span>',
+        Severity.error: '<span style="color: red;">&#10008;</span>',
+    }
+
+    def __init__(
+        self,
+        *args,
+        data: Optional[list[Union[ProcedureStep, PreparedComparison]]] = None,
+        **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        icon = self.style().standardIcon(self.style_icons[Severity.warning])
+        self.result_label.setPixmap(icon.pixmap(25, 25))
+        self.data = data
+
+        if data:
+            self.setup_buttons(configs=data)
+            # initialize tooltip
+            self.update_status()
+
+    def setup_buttons(self, configs, next_widget: AtefItem = None) -> None:
```

Comment:
Somehow this version of the buttons is bleeding over into the passive checks, is this expected?

![image](https://user-images.githubusercontent.com/10647860/224185291-204c8d1a-6e29-4840-a1a7-d776b08371dc.png)

Reply:
hm, it is not.  I think this has to do with AnyComparison being weird.  I modify the RunCheck widget based on the config associated with the widget, and I can't find a way to get the proper configs/comparisons for AnyComparison (or haven't found a clean way, at least).

I'll look into ir more

Reply:
:fire: AnyComparison :fire: strikes again